Sources/core/objects/Human.cpp

```diff
@@ -385,6 +396,54 @@ void Human::CalculateVisible(std::list<point>* visible_list)
     }
 }
 
+void Human::Bump(id_ptr_on<IMovable> item)
+{
+    if(id_ptr_on<Projectile> p = item)
+    {
+        bool damaged = false;
+        if (p.valid())
+        {
+            health_ -= p->GetDamage();
+            if (id_ptr_on<Bullet> b = p)
+            {
+                 GetGame().GetChat().PostSimpleText(name + " got hit by a bullet!", owner->GetId());
+                 unsigned int value = GetRand() % 3;
+                 std::string snd;
+                 if (value == 0)
+                     snd = "genhit1.ogg";
+                 if (value == 1)
+                     snd = "genhit2.ogg";
+                 if (value == 2)
+                     snd = "genhit3.ogg";
+                 PlaySoundIfVisible(snd, owner.ret_id());
+            }
+            damaged = true;
+        }
+        if (!damaged)
+        {
+            return;
+        }
+
+        if ((GetRand() % 3) != 0)
+        {
+            return;
+        }
+
+        unsigned int blood_value = (GetRand() % 7) + 1;
+        std::stringstream conv;
+        conv << "floor" << blood_value;
+
+        if (id_ptr_on<Floor> f = GetTurf())
+        {
+            if (!f->bloody)
+            {
+                f->GetView()->AddOverlay("icons/blood.dmi", conv.str());
+                f->bloody = true;
+            }
+        }
+    }
```

Comment:
You forget `return` here. Projectiles should not (at least, ordinary projectiles) trigger `IMovable::Bump`.